### PR TITLE
FactoryBot Migration

### PR DESCRIPTION
**Pull request by defndaines**

### CODE REVIEW

File: spec/factories/allscripts_unity_client_parameters_factory.rb [hunk 1]
```diff
@@ -12,5 +12,7 @@
     ca_file nil
     ca_path nil
     timeout nil
+
+    skip_create
```
Comment (defndaines) [inline spec/factories/allscripts_unity_client_parameters_factory.rb]: The `skip_create` allowed resolved an issue with 4.4->4.5 FactoryGirl migration. If there is a better way, let me know, but this solved the linting problem and still passes tests.

File: lib/allscripts_unity_client/client_options.rb [hunk 2]
```diff
@@ -38,10 +38,10 @@ def initialize(options = {})
...
     # See #initialize.
     def validate_options(options = {})
-      base_unity_url = options.has_key?(:base_unity_url) ? options[:base_unity_url] : @base_unity_url
-      username = options.has_key?(:username) ? options[:username] : @username
-      password = options.has_key?(:password) ? options[:password] : @password
-      appname = options.has_key?(:appname) ? options[:appname] : @appname
+      base_unity_url = options.key?(:base_unity_url) ? options[:base_unity_url] : @base_unity_url
+      username = options.key?(:username) ? options[:username] : @username
+      password = options.key?(:password) ? options[:password] : @password
+      appname = options.key?(:appname) ? options[:appname] : @appname
```
Comment (noprompt) [inline lib/allscripts_unity_client/client_options.rb]: We might as well change these to `options.fetch(:<name>, @<name>)`.

**Comment by noprompt** [comment]
@defndaines I left one suggestion but other than that this LGTM.

File: lib/allscripts_unity_client/client_options.rb (hunk 2 above)
Comment (defndaines) [reply lib/allscripts_unity_client/client_options.rb]: Thanks, cleaned that up.

**Pull request merged**